src/main/java/com/nylas/Calendars.java

```diff
@@ -73,6 +87,24 @@ public long count(CalendarQuery query) throws IOException, RequestFailedExceptio
 		return client.executePost(authUser, url, params, JsonHelper.listTypeOf(FreeBusy.class));
 	}
 
+	/**
+	 * Check calendar free or busy status
+	 * @param query The free/busy query
+	 * @return The free/busy timeslots
+	 */
+	public List<FreeBusy> checkFreeBusy(FreeBusyQuery query) throws IOException, RequestFailedException {
+		if(!query.isValid()) {
+			throw new IllegalArgumentException("Free Busy query missing one or more required parameters.");
+		}
```

Comment:
It may be easier to put this exception throw in a Query.valdiate() method that calls isValid internally and then you can put the missing params in the exception message

Reply:
and maybe add final to the parameters so those inputs are immutable by default

Reply:
Thanks, I created a validate method and you've motivated me to make better error messages! Regarding making the inputs immutable by default causes headaches in setting unless I create a separate class for the builder